assets/js/carouselClass.js

```diff
@@ -0,0 +1,105 @@
+export class CarouselClass {
+  constructor(container) {
+    // Created variables
+    this.container = container;
+    this.carouselList = this.container.querySelector(".carousel-list"),
+    this.carouselItems = this.container.querySelectorAll(".carousel-list__item"),
+    this.carouselWrap = this.container.querySelector(".carousel-wrap"),
+    this.btnNext = this.container.querySelector(".carousel-next"),
+    this.btnPrev = this.container.querySelector(".carousel-prev"),
+    this.pagination = this.container.querySelectorAll(".pagination-bullet");
+
+    // Calculated variables
+    this.carouselWrapWidth = this.carouselWrap.offsetWidth,
+    this.carouselItemsLength = this.carouselItems.length,
+    this.carouselItemsActiveIndex = 0,
+    this.isResizing = false;
+
+
+  }
+
+  // Add or disable switch transition
+  addTransition() {
+    return (this.carouselList.style.transition = this.isResizing
+      ? ""
+      : "transform 0.5s ease-in-out");
+  };
+
+  // Set active class to carousel item
+  addActiveToItem() {
+    this.carouselItems.forEach((item, index) => {
+      return index ===  this.carouselItemsActiveIndex
+        ? item.classList.add("carousel-list__item--active")
+        : item.classList.remove("carousel-list__item--active");
+    });
+  };
+
+  // Add/remove class to pagination bullet
+  setActivePagination() {
+    this.pagination.forEach((item, index) => {
+      return index === this.carouselItemsActiveIndex
+        ? item.classList.add("pagination-bullet--active")
+        : item.classList.remove("pagination-bullet--active");
+    });
+  };
+
+  // Navigate carousel on pagination click
+  moveCarouselOnPagination() {
+    this.pagination.forEach((item, index) => {
+      item.addEventListener("click", () => {
+        this.carouselItemsActiveIndex = index;
+        this.moveCarousel();
+      });
+    });
+  };
+
+  // On window resize change carouselWrapWidth to atjust transform
+  changeSize() {
+    this.carouselWrapWidth = this.carouselWrap.offsetWidth; // HERE
+    this.isResizing = true;
+    this.carouselList.style.transition = "";
+    this.moveCarousel();
+  };
+
+  // Move carousel to selected item
+  moveCarousel() {
+    let transformValue = this.carouselItemsActiveIndex * this.carouselWrapWidth;
+    this.carouselList.style.transform =
+      "translate3d(-" + transformValue + "px, 0px, 0px)";
+    this.addActiveToItem();
+    this.addTransition();
+    this.setActivePagination();
+    this.setButtonAttr();
+  };
+
+  // Move carousel to direction
+  moveTo(direction) {
+    const isFirst = this.carouselItemsActiveIndex === 0;
+    const isLast = this.carouselItemsActiveIndex === this.carouselItemsLength - 1;
```

Comment:
This logic is also reused in `setButtonAttr`. Consider using getters:
```
get isFirstItemActive() {
  return this.carouselItemsActiveIndex === 0;
}
```

and then

```
if (direction === 'forward' && !this.isFirstItemActive) { ... }
// ...
this.btnPrev.disabled = this.isFirstItemActive;
```

Reply:
It seems I cant use **get** inside class method?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#prototype_methods
You should be able to use `get` inside a class alongside any other methods or static methods.

Reply:
**alongside any other methods** gotcha! I thought you want me to set getter inside method.